### PR TITLE
[WIP][HHVM] Fixes

### PR DESCRIPTION
**Pull request by Maks3w**
Anyone is invited to send patches to this branch
- [x]  Cast SimpleXMLElement to array. `Invalid operand type was used: expecting an array` when use `SimpleXMLElement` as parameter of `each` function. facebook/hhvm/issues/1491 [XmlRpc]
- [ ] PHPUnit run in separate process feature don't work. sebastianbergmann/phpunit/issues/1090 [Session, Version] - depends on #6029 
- [x] DateTime::createFromFormat not return false when error exists. facebook/hhvm#1495
- [x] Wrong resolution of constants in default parameters value. facebook/hhvm/issues/1499 [Code]
- [x] facebook/hhvm/issues/1500 [View]
- [ ] Replace function stream_filter_append. HHVM won't implement it. [Mime]
- [x] ircmaxell/random-lib not compatible. [Crypt] https://github.com/ircmaxell/RandomLib/commit/a3c227c8b3e7c0b880762236e8a22d3ba8892489
- [x] https://github.com/facebook/hhvm/issues/1540
- [ ] https://github.com/facebook/hhvm/issues/1541
- [ ] https://github.com/facebook/hhvm/issues/1395 https://github.com/facebook/hhvm/issues/406 [Cache]
- [x] https://github.com/facebook/hhvm/pull/1542 [Filter]
- [ ] https://github.com/facebook/hhvm/issues/1552 - MVC (won't be fixed by HHVM)
- [ ] https://github.com/facebook/hhvm/issues/1553 [Config]
- [x] https://github.com/facebook/hhvm/issues/1555
- [x] mikey179/vfsStream https://github.com/facebook/hhvm/issues/1476
- [ ] https://github.com/facebook/hhvm/issues/1561 (won't be fixed by HHVM)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ language: php
 
 php:
   - 5.3
-  - 5.4
-  - 5.5
```
Comment (weierophinney) [inline .travis.yml]: Why are the 5.4 and 5.5 versions removed?

Comment (Maks3w) [inline .travis.yml]: For speed up WIP removing less important scenarios. The commit has the title DON'T MERGE

Comment (Maks3w) [inline .travis.yml]: The idea was restart the travis build for catch latest HHVM versions and monitor the progress,

**Comment by marc-mabe** [comment]
[Cache] facebook/hhvm#1395 facebook/hhvm#406 are closed by HHVM

**Comment by weierophinney** [comment]
Considering that much of this is reliant on upstream changes in HHVM, we cannot pin this to a release. We can address after 2.4 on a component-by-component basis.

**Comment by Maks3w** [comment]
Exactly. This PR does not intent fix anything. (Since any fix for HHVM should be valid for PHP). The PR is for monitor the parity/compatibility progress.

**Comment by fredemmott** [comment]
I just looked through the still-todo list:

- PHPUnit runInSeparateProcess has been working for most of the last year
- stream_filter_append() has been implemented
- APCIterator has been implemented (#1395 and #406)

Reopened: facebook/hhvm#1553 (DOMNode::appendChild)

**Comment by Maks3w** [comment]
@fredemmott Any plan for include (again) ZF2 to HHVM parity tests?

**Comment by fredemmott** [comment]
They'll be re-added when they stop taking well over an hour to run;  I don't know yet if that's due to HHVM, our test infrastructure, or ZF2, but I should have some time this week to look at this.

**Comment by Maks3w** [comment]
I'll close this. Every component will have their own roadmap with ZF2.5

**Pull request closed without merging**